dali/sasha/sadbghk.cpp

```diff
@@ -0,0 +1,180 @@
+#include "platform.h"
+
+#include "jfile.hpp"
+#include "jlib.hpp"
+#include "jlog.hpp"
+#include "jiface.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+
+#include "saserver.hpp"
+#include "sautil.hpp"
+
+static constexpr unsigned defaultExpiryDays = 14;
+static constexpr unsigned defaultHouseKeepingIntervalHours = 24;
+
+#define LOGDBGHK "DEBUGPLANEHOUSEKEEPING: "
+
+// Debug Housekeeping monitor
+
+class CSashaDebugPlaneHousekeepingServer : public ISashaServer, public Thread
+{
+    std::atomic<bool> stopped{false};
+    Semaphore stopSem;
+    Mutex runMutex;
+    StringBuffer debugDir;
+    unsigned expiryDays{0};
+
+public:
+    IMPLEMENT_IINTERFACE_USING(Thread);
+
+    CSashaDebugPlaneHousekeepingServer()
+        : Thread("CSashaDebugPlaneHousekeepingServer")
+    {
+    }
+
+    virtual void start() override
+    {
+        expiryDays = getComponentConfigSP()->getPropInt("@expiryDays", defaultExpiryDays);
+        if (expiryDays == 0)
+        {
+            WARNLOG("Failed to get expiry days for debug plane");
+            return;
+        }
+
+        // get debug plane dir
+        StringBuffer planeName;
+        if (!getDefaultPlane(planeName, "@debugPlane", "debug"))
+        {
+            WARNLOG("Failed to get default debug plane");
+            return;
+        }
+        Owned<IPropertyTree> plane = getStoragePlane(planeName);
+        assertex(plane);
+        verifyex(plane->getProp("@prefix", debugDir));
+
+        Thread::start(false);
+    }
+
+    virtual void ready() override
+    {
+    }
+
+    virtual void stop() override
+    {
+        if (!stopped)
```

Comment:
I know this is c&p from other code, but if already stopped, why continue..
I think this should be:
```
if (stopped)
    return;
stopped = true;
stopSem.signal();
....
```